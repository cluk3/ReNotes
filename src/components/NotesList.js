import React, { Component } from "react";

class NotesList extends Component {
  render() {
    return <div />;
  }
}

export default NotesList;
