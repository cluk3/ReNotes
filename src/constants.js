export const ENTITIES = {
  FOLDERS: "FOLDERS",
  NOTES: "NOTES"
};
