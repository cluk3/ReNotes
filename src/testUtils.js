import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { initialState as notesInitialState } from './Notes/modules/notes';
import { initialState as foldersInitialState } from './Folders/modules/folders';
import { initialState as focusedElementInitialState } from './containers/Focusable/stateManager';
import { DndProvider } from 'react-dnd-cjs';
import HTML5Backend from 'react-dnd-html5-backend-cjs';

const reducer = state => state;
export const defaultInitialState = {
  folders: foldersInitialState,
  notes: notesInitialState,
  focusedElement: focusedElementInitialState
};

export const wrapWithDnD = ui => (
  <DndProvider backend={HTML5Backend}>{ui}</DndProvider>
);

export function renderWithRedux(
  ui,
  {
    initialState = defaultInitialState,
    store = createStore(reducer, initialState)
  } = {}
) {
  return {
    ...render(<Provider store={store}>{wrapWithDnD(ui)}</Provider>),
    // adding `store` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    store
  };
}

export const renderWithDnD = ui =>
  render(<DndProvider backend={HTML5Backend}>{ui}</DndProvider>);
