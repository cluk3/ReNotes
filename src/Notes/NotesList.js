import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { setActiveNote, deleteNote } from './modules/notes';
import Note from './Note';
import { ENTITIES } from 'constants.js';

const NotesListContainer = styled.div`
  max-height: 100%;
  overflow-y: auto;
`;

export class NotesList extends PureComponent {
  static propTypes = {
    notes: PropTypes.array.isRequired,
    activeFolderId: PropTypes.string,
    activeNote: PropTypes.string,
    setActiveNote: PropTypes.func.isRequired,
    deleteNote: PropTypes.func.isRequired
  };

  handleNoteClick(noteId) {
    noteId !== this.props.activeNote && this.props.setActiveNote(noteId);
  }

  render() {
    const { notes, activeNote, isNoteListFocused } = this.props;
    const notesList = notes.map(({ lastModified, editorState, noteId }) => {
      const isNoteSelected = activeNote === noteId;
      return (
        <Note
          lastModified={lastModified}
          text={editorState.text}
          key={noteId}
          handleNoteClick={() => this.handleNoteClick(noteId)}
          selected={isNoteSelected}
          highlighted={isNoteListFocused && isNoteSelected}
        />
      );
    });

    return <NotesListContainer>{notesList}</NotesListContainer>;
  }
}

function mapStateToProps({ notes, folders, focusedElement }) {
  const activeFolder = folders.byId[folders.activeFolder];
  // right now since we recreate the notes everytime the component is re-rendered.
  // Can be solved by reselect memoization
  return {
    notes: activeFolder
      ? activeFolder.notes.map(noteId =>
          Object.assign({ noteId }, notes.byId[noteId])
        )
      : [],
    activeFolderId: folders.activeFolder,
    activeNote: notes.activeNote,
    isNoteListFocused: focusedElement.elementType === ENTITIES.NOTES
  };
}

const mapDispatchToProps = {
  setActiveNote,
  deleteNote
};

export default connect(mapStateToProps, mapDispatchToProps)(NotesList);
