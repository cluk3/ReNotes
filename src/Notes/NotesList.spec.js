import React from 'react';
import NotesList from './NotesList';
import { ENTITIES } from '../constants';
import { renderWithRedux } from 'testUtils';
import 'jest-styled-components';

describe('NotesList', () => {
  test('it works', () => {
    const { container } = renderWithRedux(<NotesList />);
    expect(container.firstChild).toMatchSnapshot();
  });

  // it("should render correctly", () => {
  //   const output = shallow(<NotesList {...props} />);
  //   expect(shallowToJson(output)).toMatchSnapshot();
  // });

  // describe("when Note is clicked", () => {
  //   it("should set item to delete", () => {
  //     const output = shallow(<NotesList {...props} />);
  //     output
  //       .find("Note")
  //       .first()
  //       .props()
  //       .handleNoteClick();
  //     expect(props.setItemToDelete).toHaveBeenCalledWith(
  //       ENTITIES.NOTES,
  //       props.notes[0].noteId
  //     );
  //   });

  //   it("should set active note", () => {
  //     const output = shallow(<NotesList {...props} />);
  //     output
  //       .find("Note")
  //       .first()
  //       .props()
  //       .handleNoteClick();
  //     expect(props.setActiveNote).toHaveBeenCalledWith(props.notes[0].noteId);
  //   });
  // });

  // describe("when New Note button is clicked", () => {
  //   it("should create a new note", () => {
  //     const output = shallow(<NotesList {...props} />);
  //     output
  //       .find("button")
  //       .first()
  //       .simulate("click");
  //     expect(props.createNewNote).toHaveBeenCalledWith(
  //       "testFolder",
  //       expect.stringContaining("note-")
  //     );
  //   });

  //   it("should set item to delete", () => {
  //     const output = shallow(<NotesList {...props} />);
  //     output
  //       .find("button")
  //       .first()
  //       .simulate("click");
  //     expect(props.setItemToDelete).toHaveBeenCalledWith(
  //       ENTITIES.NOTES,
  //       expect.stringContaining("note-")
  //     );
  //   });

  //   it("should set active note", () => {
  //     const output = shallow(<NotesList {...props} />);
  //     output
  //       .find("button")
  //       .first()
  //       .simulate("click");
  //     expect(props.setActiveNote).toHaveBeenCalledWith(
  //       expect.stringContaining("note-")
  //     );
  //   });
  // });
});
