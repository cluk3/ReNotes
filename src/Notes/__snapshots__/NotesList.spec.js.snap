// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesList it works 1`] = `
.c0 {
  max-height: 100%;
  overflow-y: auto;
}

.c1 {
  background-color: #dfdfdd;
  padding-left: 1em;
}

.c7 {
  background-color: #f9f9f7;
  padding-left: 1em;
}

.c2 {
  font-size: 14px;
  padding: 1em 1em 1em 0;
  border-bottom: #dedede solid 1px;
}

.c3 {
  font-size: 15px;
  margin: 0;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  width: 100%;
  line-height: 1.4;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
}

.c5 {
  padding-right: 1em;
  white-space: nowrap;
}

.c6 {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h2
        class="c3"
      >
        lorem ipsum foo bar sit dolor amen lorepsum foo bar sit dolor amen
      </h2>
      <div
        class="c4"
      >
        <span
          class="c5"
        >
          Mocked
        </span>
        <span
          class="c6"
        >
          No additional text
        </span>
      </div>
    </div>
  </div>
  <div
    class="c7"
  >
    <div
      class="c2"
    >
      <h2
        class="c3"
      >
        lorem ipsum foo bar sit
      </h2>
      <div
        class="c4"
      >
        <span
          class="c5"
        >
          Mocked
        </span>
        <span
          class="c6"
        >
          No additional text
        </span>
      </div>
    </div>
  </div>
</div>
`;
